src/sql/finalize.sql

```diff
@@ -727,6 +729,10 @@ CREATE OPERATOR CLASS veci8_cos_ops
     FOR TYPE veci8 USING vectors AS
     OPERATOR 1 <=> (veci8, veci8) FOR ORDER BY float_ops;
 
+CREATE OPERATOR CLASS bigint_attr_ops
+    DEFAULT FOR TYPE bigint USING vectors AS
+    OPERATOR 1 = (bigint, bigint);
```

Comment:
We should add a function here to convert any datatypes to an integer.

Reply:
> If neither `FOR SEARCH` nor `FOR ORDER BY` is specified, `FOR SEARCH` is the default.

Maybe `FOR SEARCH` is better than `FOR ORDER BY`, because the results returned by the backend are not sorted by the second column.

Reply:
If we take advantage of `OPERATOR FOR SEARCH`, maybe we will lost some chances about optimization, for example:

* `SELECT * FROM t ORDER BY embedding <-> '[0, 0, 0]' WHERE label = 'alpha' OR label = 'beta'`. It's a classical HNSW query (see https://qdrant.tech/articles/filtrable-hnsw/, "Select points which belong to a specific subset of categories"). But `label = 'alpha' OR label = 'beta'` will be a filter cond, not an index clause. It can be solved if we provide a new operator `<:` and force users to write `label <: ('label', 'beta')`.
* `SELECT * FROM t ORDER BY embedding <-> '[0, 0, 0]' WHERE id = id`. `id = id` is also a filter cond, not an index clause. PostgreSQL will fetch rows in table. But if it's checked by an index, it's faster. It can be solved if we support index only scan...

Reply:
I did some investigation last week, on how btree/heap index filter about the where condition. It seems we can reuse this part of the code like https://github.com/postgres/postgres/blob/master/src/backend/access/heap/heapam.c#L792-L793. Is it necessary for us to define a new operator class here?

Reply:
> Is it necessary for us to define a new operator class here?

How if a new operator is defined here? `ambuild/aminsert` get column data, and shouldn't the index need to known it's an integer, string or something else?

Reply:
I'm not sure. But does btree or other postgres need a new operator type? I think the case here should be similar.

Reply:
GIN/GIST index also support data type like tsvector. How does they know the real data and push down the filter condition to the index scan?